src/core/shape_inference/include/extract_image_patches_shape_inference.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/extractimagepatches.hpp>
+
+namespace ov {
+namespace op {
+namespace v3 {
+template <class T>
+void shape_infer(const ExtractImagePatches* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
+    const auto& input_shape = input_shapes[0];
+    auto& output_shape = output_shapes[0];
+    output_shape.resize(4);
+    NODE_VALIDATION_CHECK(op, input_shape.rank().compatible(4), "input tensor must be 4D tensor.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_sizes.size() == 2,
+                          "Attribute sizes should be in [size_rows, size_cols] format.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_movement_strides.size() == 2,
+                          "Attribute strides should be in [stride_rows, stride_cols] format.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_movement_strides[0] > 0 && op->m_patch_movement_strides[1] > 0,
+                          "Attribute strides should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_selection_rates.size() == 2,
+                          "Attribute rates should be in [rate_rows, rate_cols] format.");
+
+    NODE_VALIDATION_CHECK(op,
+                          op->m_patch_selection_rates[0] > 0 && op->m_patch_selection_rates[1] > 0,
+                          "Attribute rates should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(
+        op,
+        op->m_padding == PadType::VALID || op->m_padding == PadType::SAME_LOWER || op->m_padding == PadType::SAME_UPPER,
+        "Attribute padding should be in either valid or same_lower or same_upper.");
+    // Determine Batch
+    output_shape[0] = input_shape[0];
```

Comment:
We still refer to a dimension with a potentially dynamic `input_shape` rank here. Could you please correct it?

Reply:
@v-Golubev But the `input_shape[0]` is the batch dimension, no matter whether it's dynamic, we could still set it. This assignment covers two conditions, firstly the batch is dynamic, then the output_shape should be dynamic, secondly, if the batch is known, then we should also set it.

Reply:
I mean that we can't refer to PartialShape elements in case when rank.is_dynamic() (not batch dimension). This code doesn't handle the following situation:
```
auto input_shape = ov::PartialShape::dynamic();
...
output_shape[0] = input_shape[0]; // Exception: "Accessing out-of-range dimension in Dimension[]"
```

Reply:
That's great, thank you!

Reply:
fix in dade4b50e6ca10b519b2891eb81a26b920773bff